### ISSUE TITLE
Simple Conceptual Questions

### ISSUE THREAD

**Issue by SeanPedersen** [guillaume-chevalier/Spiking-Neural-Network-SNN-with-PyTorch-where-Backpropagation-engenders-STDP#7]
Hey first: great repo with clear and detailed explanations!

I am just wondering: 
- From what I gathered SNN are learning in the time domain (building / fading firing potentials), so I assume the ordering of the training sample matters to the algorithm? How would for example the final learned weights change between training on a randomly shuffled data set and a per class sorted data set? I would imagine it having a big influence, curious for an answer.
- If the order of data is important for training, is it for testing / inference as well?

Hope my questions make sense.

**Comment by guillaume-chevalier** [comment]
Hi @SeanPedersen ! 

For what regards the order of data, each training sample (as well as test/prod samples) are fed throughout a certain duration: see the `for _ in range(self.n_time_steps):` part in the code for this duration loop.

This way, each training data is sent in an isolated fashion such that the activations and the hidden accumulated values are reset between each training example. I'd recommend shuffling your dataset like you would treat any other Artificial Neural Network as per the fact that the network's activations are reset between each example, thus not leaving a specific trace behind each time (other than the learned/updated weighs). 

To sum up, I'd recommend shuffling data like any other neural net. You could also do curriculum learning as on any other neural net. 

Best, 
G
